Source/Forms/Gen4/Tools/Poketch.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Poketch.hpp"
+#include "ui_Poketch.h"
+#include <Core/Util/Utilities.hpp>
+
+Poketch::Poketch(u32 advances, QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::Poketch)
+{
+    ui->setupUi(this);
+    setAttribute(Qt::WA_QuitOnClose, false);
+    setAttribute(Qt::WA_DeleteOnClose);
+
+    setupModels(advances);
+
+}
+
+void Poketch::setupModels(u32 advances) {
+
+    connect(ui->pushButtonClose, &QPushButton::clicked, this, &Poketch::on_pushButtonClose_clicked);
+
+    u32 coin = 0;
+    u32 happy = 0;
+    u32 target = 0;
+
+    if(advances < 12)
+    {
+        coin = advances;
+        ui->labelNote->setText("DO NOT SWITCH TO THE HAPPINESS APPLICATION AT ALL");
```

Comment:
Wrap in a tr() call. Is caps lock necessary?

Reply:
Caps lock just cause it was in Reporter lol fixed though